### DISCUSSION TITLE
Hololens 2: How to occlude virtual objects with dynamic real-world objects (i.e., hands)?

### DISCUSSION THREAD

**Discussion by A-Ivan** [microsoft/MixedRealityToolkit-Unity#10481] [Q&A] [answered]
Hi,

Are there any suggestions on how to occlude (mask out) virtual objects with dynamic real-world objects (user's hands or other moving objects) on the Hololens 2? 

For instance, an object appears 2 meters in front of the Hololens (floating above the ground) and if I move my hands, or if another person passes (more than 1 meter away from the Hololens), between the Hololens and the virtual object, I would expect the virtual object to be occluded, but this does not occur. 

Would [spatial awareness](https://docs.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/features/spatial-awareness/spatial-awareness-getting-started?view=mrtkunity-2021-05) need to be turned on to occlude virtual objects in these cases with dynamic real-world objects? From what I understood, spatial mapping can be used to occlude virtual objects, but the real-world environment needs to be scanned, which is not useful if the occlusion comes from moving objects.

I would expect that the user's hands would at least be able to occlude objects, since it can have a [virtual representation/mesh](https://docs.microsoft.com/en-us/windows/mixed-reality/mrtk-unity/features/input/hand-tracking?view=mrtkunity-2021-05#hand-mesh-prefab) when tracked. I may have missed something in the documents.

Thank you,
Ivan

**Comment by Zee2** [comment, marked as answer]
Hi @A-Ivan;

We generally discourage folks from using hand occlusion on HoloLens 2 because the hand tracking latency isn't quite up to snuff to be able to pull off a convincing occlusion effect.

For non-hand moving objects, there is no support for occlusion at all because there is no capability for tracking + estimating depth of dynamic real-world objects, unfortunately!

Environmental occlusion is trivial, though! Just use the included "occlusion" material on the SR mesh.
